### PR TITLE
DEV: upgrade insert video modal

### PR DESCRIPTION
**Pull request by tyb-talks**
### Context
This updates the insert video modal to use the DModal API.

Not doing:  This change does not include fixing the deprecation warnings that exist with the current modal - this would involve changing the `poster-uploader` / `custom-simple-list` components, and is out of scope here.

Test artifact:
https://github.com/discourse/discourse-insert-video/assets/133760061/0c1fba08-2e2d-46d1-b957-937ae093ab33

### CODE REVIEW

**Review by davidtaylorhq — APPROVED** [review]

File: common/common.scss [hunk 1]
```diff
@@ -18,10 +18,8 @@
   }
 }
 
-.modal-inner-container {
-  .insert-video-inputs .select-kit {
-    width: 450px;
-  }
+.d-modal.insert-video-modal .d-modal__container {
+  width: 100%;
```
Comment (CvX) [inline common/common.scss:22]: Isn't that the default?

File: javascripts/discourse/components/modal/insert-video.hbs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<DModal
+  @closeModal={{@closeModal}}
+  @bodyClass="insert-video"
```
Comment (CvX) [inline javascripts/discourse/components/modal/insert-video.hbs:3]: Is anything using this class?

File: javascripts/discourse/components/modal/insert-video.hbs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    {{/if}}
+  </:body>
+  <:footer>
+    <DButton
+      class="btn-primary"
+      data-test-id="insert-video-button"
+      @disabled={{this.insertDisabled}}
+      @label={{theme-prefix "modal.insert"}}
+      @action={{this.insertVideo}}
+    />
```
[the remark this answers is not in the document]
Comment (CvX) [reply javascripts/discourse/components/modal/insert-video.hbs:46]: No new/updated code should use `{{component-name}}` syntax anyway!

File: javascripts/discourse/components/modal/insert-video.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  _preventSubmit() {
+    // prevent submitting on enter while adding items to lists using Enter
+    document
+      .querySelector(".video-sources")
+      .addEventListener("keydown", (e) => this._keyDown(e));
+
+    document
+      .querySelector(".video-subtitles")
+      .addEventListener("keydown", (e) => this._keyDown(e));
+  }
```
Comment (CvX) [inline javascripts/discourse/components/modal/insert-video.js:32]: Does this work in a glimmer component? 🤔 even if it does, imo it'd be better to use `{{did-insert}}` modifier to run this.

File: javascripts/discourse/components/modal/insert-video.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  @afterRender
+  _preventSubmit() {
+    // prevent submitting on enter while adding items to lists using Enter
+    document
+      .querySelector(".video-sources")
+      .addEventListener("keydown", (e) => this._keyDown(e));
+
+    document
+      .querySelector(".video-subtitles")
+      .addEventListener("keydown", (e) => this._keyDown(e));
```
Comment (CvX) [inline javascripts/discourse/components/modal/insert-video.js:31]: 🏌️

```suggestion
      .addEventListener("keydown", this._keyDown);

    document
      .querySelector(".video-subtitles")
      .addEventListener("keydown", this._keyDown);
```

**Pull request merged**